packages/io/amf-deserializer/translate.js

```diff
@@ -2,93 +2,86 @@ const createObject = require('./objectBuilder')
 const parse = require('./parse')
 
 const translate = function (src, filename, options) {
-  options && options.statusCallback && options.statusCallback({progress: 0})
-  filename = filename || 'amf'
-  const defaults = {pxPmm: require('./constants').pxPmm, version: '0.0.0', addMetaData: true}
+  const defaults = {
+    pxPmm: require('./constants').pxPmm
+  }
   options = Object.assign({}, defaults, options)
-  const {version, pxPmm, addMetaData} = options
+  const { version, pxPmm, addMetaData } = options
+
+  options && options.statusCallback && options.statusCallback({ progress: 0 })
 
   // parse the AMF source
-  const {amfObj, amfMaterials, amfTextures, amfConstels} = parse(src, pxPmm)
+  const { amfObj, amfMaterials, amfTextures, amfConstels } = parse(src, pxPmm)
+
   // convert the internal objects to JSCAD code
   let code = addMetaData ? `//
-  // producer: OpenJSCAD.org ${version} AMF deserializer
-  // date: ${new Date()}
-  // source: ${filename}
-  //
-  ` : ''
+// producer: OpenJSCAD.org ${version} AMF deserializer
+// date: ${new Date()}
+// source: ${filename}
+//
+` : ''
 
   if (!amfObj) {
     throw new Error('AMF parsing failed, no valid AMF data retrieved')
   }
 
-  options && options.statusCallback && options.statusCallback({progress: 50})
-  
-  const scadCode = codify(amfObj, {amfMaterials, amfTextures, amfConstels})
-  code += scadCode
-  
-  options && options.statusCallback && options.statusCallback({progress: 100})
+  options && options.statusCallback && options.statusCallback({ progress: 50 })
+
+  code += codify(amfObj, { amfMaterials, amfTextures, amfConstels })
+
+  options && options.statusCallback && options.statusCallback({ progress: 100 })
+
   return code
 }
 
 //
 // convert the internal repreentation into JSCAD code
 //
-function codify (amf, data) {
+const codify = (amf, data) => {
   if (amf.type !== 'amf' || (!amf.objects)) throw new Error('AMF malformed')
+
   let code = ''
 
   // hack due to lack of this in array map()
   let objects = amf.objects
   let materials = data.amfMaterials
 
   // convert high level definitions
-  function createDefinition (obj, didx) {
-    // console.log(materials.length);
-    switch (obj.type) {
+  // this ~= data
+  const createDefinition = (object, index) => {
+    switch (object.type) {
       case 'object':
-        code += createObject(obj, didx, data, {csg: false})
+        code += createObject(object, index, data, { csg: false, scale: amf.scale })
```

Comment:
much clearer !

Reply:
hmm csg: false should probably be 'geometry: false' ? what do you think ?

Reply:
Done